### DISCUSSION TITLE
Demande relative à la syntaxe !?

### DISCUSSION THREAD

**Discussion by simogeo** [publicodes/publicodes#398] [Q&A]
Bonjour, je crossposte ici une question relative à la syntaxe publicodes plus qu'à NGC que j'adapte pour une déclinaison locale : 

https://github.com/datagir/nosgestesclimat-site/issues/975

Merci pour vos lumières !

**Comment by johangirod** [comment]
Bonjour. Ce doit être lors de l'appel à `setSituation` que se situe l'erreur.
Au lieu de  : 

```js 
engine.setSituation({
  'logement . résidence': 'Tahiti'
})
```
Il faudra faire attention à écrire : 
```js 
engine.setSituation({
  'logement . résidence': "'Tahiti'"
})
```

**Comment by simogeo** [comment]
Merci @johangirod

Pour bien comprendre, l'erreur devrait, selon toi, être corrigée ans le code source lui-même ou  ou dans le modèle de données (/data/) ? 

Je viens de faire un test sur ce dernier mais sans succès ...

Pour info, j'utilise une version de NGC qui date de juin 2022 environ. Je crois que le moteur a changé depuis. 

Mon fichier _EngineContext.tsx_ contient entre autre : 


```
export function SituationProvider({
	children,
	situation,
}: SituationProviderProps) {
	const engine = useContext(EngineContext)
	engine.setSituation(situation)
	return (
		<EngineContext.Provider value={engine}>{children}</EngineContext.Provider>
	)
}
```

**Comment by simogeo** [comment]
Ok, je progresse dans mes investigations. Quand on est sur une présentation radio button, ca passe (5 items) mais ca ne passe plus dès qu'il y a plus de 6 éléments et un passage en dropdown / select ....
